test/Microsoft.AspNetCore.Authentication.Test/OpenIdConnect/OpenIdConnectEventTests.cs

```diff
@@ -1669,41 +1024,41 @@ public class OpenIdConnectEventTests
         [Fact]
         public async Task OnRemoteSignOut_Handled_NoSignout()
         {
-            var forSignOut = false;
-            var server = CreateServer(CreateNotImpEvents(events =>
+            var events = new ExpectedOidcEvents()
             {
-                events.OnRemoteSignOut = context =>
-                {
-                    forSignOut = true;
-                    context.Response.StatusCode = StatusCodes.Status202Accepted;
-                    context.HandleResponse();
-                    return Task.CompletedTask;
-                };
-            }),
-            AppNotImpl);
+                ExpectRemoteSignOut = true,
+            };
+            events.OnRemoteSignOut = context =>
+            {
+                events.InvokedRemoteSignOut = true;
+                context.Response.StatusCode = StatusCodes.Status202Accepted;
+                context.HandleResponse();
+                return Task.CompletedTask;
+            };
+            var server = CreateServer(events, AppNotImpl);
 
             var client = server.CreateClient();
             var response = await client.GetAsync("/signout-oidc");
 
             Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
-            Assert.True(forSignOut);
+            events.ValidateExpectations();
             Assert.False(response.Headers.TryGetValues(HeaderNames.SetCookie, out var values));
         }
 
         [Fact]
         public async Task OnRemoteSignOut_Skip_NoSignout()
         {
-            var forSignOut = false;
-            var server = CreateServer(CreateNotImpEvents(events =>
+            var events = new ExpectedOidcEvents()
             {
-                events.OnRemoteSignOut = context =>
-                {
-                    forSignOut = true;
-                    context.SkipHandler();
-                    return Task.CompletedTask;
-                };
-            }),
-            context =>
+                ExpectRemoteSignOut = true,
+            };
+            events.OnRemoteSignOut = context =>
+            {
+                events.InvokedRemoteSignOut = true;
```

Comment:
Is there any way to clean up this pattern to avoid needing to remember to set the invoked?  Maybe just add a new set of virtuals in the custom event class to override in the test...

Forgetting to set the bool will cause the validation to fail right?  Meaning that line is required always right?

Reply:
Basically something like `events.TestXyzEvent(Action<XyzContext> test)` method for the test to use instead which takes care of the InvokeXyz = true.